diskmap/diskmap.py

```diff
@@ -205,6 +210,23 @@ def power_law_height(
 
             # opening angle (rad)
             disk_opening = np.arctan2(disk_height, disk_radius)
+            
+        elif surface == "function":
+            
+            if height_func is None:
+                raise ValueError("If using surface=='function', you must specify height_func")
+            
+            # midplane radius (au)
+            disk_radius = np.linspace(radius[0], radius[1], radius[2])
+
+            # disk height (au)
+            try:
```

Comment:
Instead of try-except, it could perhaps be simplified by using the type annotation+check. Something like `Optional[Callable[np.ndarray, np.ndarray]` but please check